src/main/java/svc/data/citations/CitationDAO.java

```diff
@@ -1,128 +1,50 @@
+@Component
+public class CitationDAO {
+	@Inject
+    private CitationDataSourceFactory citationDataSourceFactory;
 
-import svc.data.jdbc.BaseJdbcDao;
-import svc.logging.LogSystem;
-import svc.models.Citation;
-import svc.util.DatabaseUtilities;
-import svc.models.Court;
-import svc.types.HashableEntity;
+	public List<Citation> getByCitationNumberAndDOB(String citationNumber, LocalDate dob) {
+	    List<CitationDataSource> sources = citationDataSourceFactory.getAllCitationDataSources();
 
-@Repository
-public class CitationDAO extends BaseJdbcDao {	
-	public Citation getByCitationId(Long citationId) {
-		try  {
-			Map<String, Object> parameterMap = new HashMap<String, Object>();
-			parameterMap.put("citationId", citationId);
-			String sql = "SELECT * FROM citations WHERE id = :citationId";
-			Citation citation = jdbcTemplate.queryForObject(sql, parameterMap, new CitationSQLMapper());
-			return citation;
-		} catch (Exception e) {
-			LogSystem.LogDBException(e);
-			return null;
-		}
-	}
-	
-	public Citation getByCitationNumber(String citationNumber) {
-		try {
-			Map<String, Object> parameterMap = new HashMap<String, Object>();
-			parameterMap.put("citationNumber", citationNumber);
-			String sql = "SELECT * FROM citations WHERE citation_number = :citationNumber";
-			Citation citation = jdbcTemplate.queryForObject(sql, parameterMap, new CitationSQLMapper());
-			return citation;
-		} catch (Exception e) {
-			LogSystem.LogDBException(e);
-			return null;
-		}
-	}
-	
-	public Citation getByCitationNumberAndDOB(String citationNumber, LocalDate dob) {
-		try {
-			Map<String, Object> parameterMap = new HashMap<String, Object>();
-			parameterMap.put("citationNumber", citationNumber);
-			parameterMap.put("dob", Date.valueOf(dob));
-			String sql = "SELECT * FROM citations WHERE citation_number = :citationNumber AND date_of_birth = :dob";
-			Citation citation = jdbcTemplate.queryForObject(sql, parameterMap, new CitationSQLMapper());
-			
-			return citation;
-		} catch (Exception e) {
-			LogSystem.LogDBException(e);
-			return null;
-		}
-	}
-	
-	public List<Citation> getByDOBAndLicense(LocalDate dob, String driversLicenseNumber) {
-		try {
-			Map<String, Object> parameterMap = new HashMap<String, Object>();
-			parameterMap.put("driversLicenseNumber", driversLicenseNumber);
-			parameterMap.put("dob", Date.valueOf(dob));
-			String sql = "SELECT * FROM citations WHERE date_of_birth = :dob AND drivers_license_number = :driversLicenseNumber";
-			List<Citation> citations = jdbcTemplate.query(sql, parameterMap, new CitationSQLMapper());
-			
-			return citations;
-		} catch (Exception e) {
-			LogSystem.LogDBException(e);
-			return null;
+        List<Observable<Citation>> citationSearches = Lists.newArrayList();
+		for(CitationDataSource source : sources) {
+            citationSearches.add(Observable.from(source.getByCitationNumberAndDOB(citationNumber, dob)).onExceptionResumeNext(Observable.just(null)));
```

Comment:
I'm actually not sure yet. It doesn't appear to handle the exception thrown inside the Tyler data source. But it doesn't blow up anything either. I think we'll have to revisit this when fully integrating the Tyler source.